src/Service.Tests/Unittests/SerializationDeserializationTests.cs

```diff
@@ -247,6 +254,63 @@ public void TestColumnDefinitionNegativeCases()
             Assert.IsFalse(_columnDefinition.Equals(col));
         }
 
+        /// <summary>
+        /// Validates serialization and deserilization of Dictionary containing DatabaseTable
+        /// this is how we serialize and deserialize metadataprovider.EntityToDatabaseObject dict
+        /// </summary>
+        [TestMethod]
+        public void TestDictionaryDatabaseObjectSerializationDeserialization()
+        {
+            InitializeObjects();
+
+            Dictionary<string, DatabaseObject> dict = new() { { "person", _databaseTable } };
+
+            string serializedDict = JsonSerializer.Serialize(dict, _options);
+            Dictionary<string, DatabaseObject> deserializedDict = JsonSerializer.Deserialize<Dictionary<string, DatabaseObject>>(serializedDict, _options)!;
+
+            DatabaseTable deserializedDatabaseTable = (DatabaseTable)deserializedDict["person"];
+
+            Assert.AreEqual(deserializedDatabaseTable.SourceType, _databaseTable.SourceType);
+            Assert.AreEqual(deserializedDatabaseTable.FullName, _databaseTable.FullName);
+            deserializedDatabaseTable.Equals(_databaseTable);
+            VerifySourceDefinitionSerializationDeserialization(deserializedDatabaseTable.SourceDefinition, _databaseTable.SourceDefinition, "FirstName");
+            VerifySourceDefinitionSerializationDeserialization(deserializedDatabaseTable.TableDefinition, _databaseTable.TableDefinition, "FirstName");
+        }
+
+        /// <summary>
+        /// Test to check the impact of version change in TypeName
+        /// </summary>
+        [TestMethod]
+        public void TestTypeNameVersionChangeImpact()
+        {
+            InitializeObjects();
+
+            Dictionary<string, DatabaseObject> dict = new() { { "person", _databaseTable } };
+
+            string serializedDict = JsonSerializer.Serialize(dict, _options);
+            string pattern = @"(Version=)\d+\.\d+\.\d+\.\d+";
+
+            // Define the replacement version number
+            string replacement = "Version=2.0.0.0";
+
+            // Use Regex.Replace to replace the version number in the JSON string
+            string modifiedJsonString = Regex.Replace(serializedDict, pattern, replacement);
+
+            Dictionary<string, DatabaseObject> deserializedDict = JsonSerializer.Deserialize<Dictionary<string, DatabaseObject>>(modifiedJsonString, _options);
+            DatabaseTable deserializedDatabaseTable = (DatabaseTable)deserializedDict["person"];
+            Assert.AreEqual(deserializedDatabaseTable.SourceType, _databaseTable.SourceType);
+
+            // Define the replacement version number
+            replacement = "Version=0.0.9";
+
+            // Use Regex.Replace to replace the version number in the JSON string
+            modifiedJsonString = Regex.Replace(serializedDict, pattern, replacement);
+
+            deserializedDict = JsonSerializer.Deserialize<Dictionary<string, DatabaseObject>>(modifiedJsonString, _options);
```

Comment:
If the version number changes, I'd expect that more aspects of the object change too. For example, when `DatabaseObject` is upgraded to a new version, there is a possibility that new properties are added. Would new properties break this (de)serialization? Wouldn't the old and new objects no longer be equal?

Reply:
yes there is a possibility of DatabaseObject having more properties, I have added check in test to compare number of properties for most of the objects I am comparing. 
Now I agree this is a bit risk as contributor needs to update the object in test and also compare that property. 

there is an idea to use reflections, but in first pass when I checked it failed. I need to spend more time in investigations but I am going to in next round.